conan/api/subapi/upload.py

```diff
@@ -99,6 +188,7 @@ def _upload_pkglist(pkglist, subtitle=lambda _: None):
             thread_pool.join()
         elapsed = time.time() - t
         ConanOutput().success(f"Upload completed in {int(elapsed)}s\n")
+        return self._add_upload_info(package_list, upload_info)
```

Comment:
Don't need to wait to add the info to this point, it can be added in-place.

Reply:
I'm going to give it a try, I was worried about this because of the upload thread pool and the package_list split.

`thread_pool.map(_upload_pkglist, package_list.split())`